File: carrito_v4/src-gen/windows/ViewModel/ProviderDAO.cs
//Start of user code imports
using CarritoDeCompras.Common;
using CarritoDeCompras.Model;
using SQLite;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
//End of user code

namespace CarritoDeCompras.ViewModels
{
    class ProviderDAO
    {
        private SQLiteConnection dbConn = new SQLiteConnection(MySQLiteHelper.DbPath);

        // Insert the new provider in the Provider table. 
        public void addProvider(Provider provider)
        {
            using (var dbConn = new SQLiteConnection(MySQLiteHelper.DbPath))
            {
                dbConn.RunInTransaction(() =>
                {
                    dbConn.Insert(provider);
                });
            }
        }

		// Retrieve the specific Provider from the database. 
        public Provider getProvider(int id)
        {
            using (dbConn)
            {
				var queryResult = dbConn.Query<Provider>("select * from Provider where idProvider =" + id).FirstOrDefault();
                return queryResult;
            }
        }

        // Retrieve the all Provider List from the database. 
        public ObservableCollection<Provider> getAllProvider()
        {
            using (dbConn)
            {
                List<Provider> myCollection = dbConn.Table<Provider>().ToList<Provider>();
                ObservableCollection<Provider> providerList = new ObservableCollection<Provider>(myCollection);
                return providerList;
            }
        }

        // Retrieve the all Provider List from the database and get the Total Rows. 
        public int getProviderCount()
        {
            using (dbConn)
            {
                 List<Provider> myCollection = dbConn.Table<Provider>().ToList<Provider>();
                ObservableCollection<Provider> providerList = new ObservableCollection<Provider>(myCollection);
                return providerList.Count;
            }
        }

        //Update existing Provider 
        public void updateProvider(Provider provider)
        {
            using (dbConn)
            {
                var queryResult = dbConn.Query<Provider>("select * from Provider where idProvider = "+ provider.idProvider).FirstOrDefault();
                if (queryResult != null)
                {
					queryResult.idProvider = provider.idProvider;		
					queryResult.ruc = provider.ruc;		
					queryResult.description = provider.description;		
					queryResult.nombre = provider.nombre;		

                    dbConn.RunInTransaction(() =>
                    {
                        dbConn.Update(queryResult);
                    });
                }
            }
        }



        //Delete specific Provider 
        public void deleteProvider(Provider provider)
        {
            using (var dbConn = new SQLiteConnection(MySQLiteHelper.DbPath))
            {
                var queryResult = dbConn.Query<Provider>("select * from Provider where idProvider = "+ provider.idProvider).FirstOrDefault();

                if (queryResult != null)
                {
                    dbConn.RunInTransaction(() =>
                    {
                        dbConn.Delete(queryResult);
                    });
                }
            }
        }
        //Delete all providerList or delete Provider table 
        public void deleteAllProvider()
        {
            using (var dbConn = new SQLiteConnection(MySQLiteHelper.DbPath))
            {
                //dbConn.RunInTransaction(() => 
                //   { 
                dbConn.DropTable<Provider>();
                dbConn.CreateTable<Provider>();
                dbConn.Dispose();
                dbConn.Close();
                //}); 
            }
        }
    }

}

File: carrito_v4/src-gen/android/app/src/main/java/com/example/carritodecompras/bd/dao/ProviderDAO.java

package com.example.carritodecompras.bd.dao;//Start of user code imports

//Start of user code imports
import android.content.ContentValues;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import com.example.carritodecompras.bd.tables.ProviderTable;
import com.example.carritodecompras.beans.Provider;
import com.example.carritodecompras.helpers.SQLiteHelper;


import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;
//End of user code
	
public class ProviderDAO {

	private SQLiteHelper mySQLiteHelper;

		public ProviderDAO(SQLiteHelper mySQLiteHelper) {
			this.mySQLiteHelper = mySQLiteHelper;
		}

		// Adding new Provider
		public void addProvider(Provider provider) {
			SQLiteDatabase db = mySQLiteHelper.getWritableDatabase();

			ContentValues values = new ContentValues();
			values.put(ProviderTable.COLUMN_DESCRIPTION, provider.getDescription().toString());		
			values.put(ProviderTable.COLUMN_IDPROVIDER, provider.getIdProvider().toString());		
			values.put(ProviderTable.COLUMN_RUC, provider.getRuc().toString());		
			values.put(ProviderTable.COLUMN_NOMBRE, provider.getNombre().toString());		
			
			// Inserting Row
			db.insert(ProviderTable.TABLE_NAME, null, values);
			db.close(); // Closing database connection
		}

		// Getting single Provider    
		public Provider getProvider(int id) {   
			SQLiteDatabase db = mySQLiteHelper.getReadableDatabase();

			Provider provider = new Provider();

			Cursor cursor = db.query(ProviderTable.TABLE_NAME, new String[] {
					ProviderTable.COLUMN_DESCRIPTION, ProviderTable.COLUMN_IDPROVIDER, ProviderTable.COLUMN_RUC, ProviderTable.COLUMN_NOMBRE
					},		ProviderTable.COLUMN_IDPROVIDER + "=?",			
					new String[] { String.valueOf(id) }, null, null, null, null);
			
			if (cursor != null) {
				cursor.moveToFirst();
				
				provider = new Provider(
						cursor.getString(0), Integer.parseInt(cursor.getString(1)), cursor.getString(2), cursor.getString(3)				
				); 

				cursor.close();
			}

			return provider;
		}


		// Getting All Provider
		public List<Provider> getAllProvider() {
			SQLiteDatabase db = mySQLiteHelper.getReadableDatabase();

			List<Provider> providerList = new ArrayList<Provider>();
			// Select All Query
			String selectQuery = "SELECT  * FROM " + ProviderTable.TABLE_NAME;

			Cursor cursor = db.rawQuery(selectQuery, null);

			// looping through all rows and adding to list
			if (cursor.moveToFirst()) {
				do {
					Provider provider = new Provider(
					cursor.getString(1), Integer.parseInt(cursor.getString(2)), cursor.getString(3), cursor.getString(4)
					); 
					
					// Adding provider to list
					 providerList.add( provider);
				} while (cursor.moveToNext());
			}

			cursor.close();

			//return provider list
			return providerList;
		}



		// Getting provider Count
		public int getProviderCount() {
			SQLiteDatabase db = mySQLiteHelper.getReadableDatabase();

			String countQuery = "SELECT  * FROM " + ProviderTable.TABLE_NAME;
			Cursor cursor = db.rawQuery(countQuery, null);

			int count = cursor.getCount();

			cursor.close();

			// return count
			return count;
		}


		// Updating single provider
		public int updateProvider(Provider provider) {
			SQLiteDatabase db = mySQLiteHelper.getWritableDatabase();

			ContentValues values = new ContentValues();
			values.put(ProviderTable.COLUMN_DESCRIPTION, provider.getDescription().toString());		
			values.put(ProviderTable.COLUMN_IDPROVIDER, provider.getIdProvider().toString());		
			values.put(ProviderTable.COLUMN_RUC, provider.getRuc().toString());		
			values.put(ProviderTable.COLUMN_NOMBRE, provider.getNombre().toString());		

			// updating row
			return db.update(ProviderTable.TABLE_NAME, values,
						ProviderTable.COLUMN_IDPROVIDER + "=?",
						new String[] { String.valueOf(provider.getIdProvider()) });		
					

		}

		// Deleting single provider
		public void deleteProvider(Provider provider) {
			SQLiteDatabase db = mySQLiteHelper.getWritableDatabase();

			db.delete(ProviderTable.TABLE_NAME,
					ProviderTable.COLUMN_IDPROVIDER + "=?",
					new String[] { String.valueOf(provider.getIdProvider()) });		

			db.close();
		}

		// Deleting all provider
		public void deleteAllProvider() {
			SQLiteDatabase db = mySQLiteHelper.getWritableDatabase();

			db.delete(ProviderTable.TABLE_NAME, null, null);
			db.close();
		}

}
